nemo/core/classes/mixins/access_mixins.py

```diff
@@ -68,15 +73,37 @@ def get_module_registry(cls, module: torch.nn.Module):
                 module_registry[name] = m._registry
         return module_registry
 
-    def reset_registry(self):
+    def reset_registry(self: torch.nn.Module, registry_key: Optional[str] = None):
         """
         Reset the registries of all named sub-modules
         """
         if hasattr(self, "_registry"):
-            self._registry.clear()
+            if registry_key is None:
+                self._registry.clear()
+            else:
+                if registry_key in self._registry:
+                    self._registry.pop(registry_key)
+                else:
+                    raise KeyError(
+                        f"Registry key `{registry_key}` provided, but registry does not have this key.\n"
+                        f"Available keys in registry : {list(self._registry.keys())}"
+                    )
+
         for _, m in self.named_modules():
             if hasattr(m, "_registry"):
-                m._registry.clear()
+                if registry_key is None:
+                    m._registry.clear()
+                else:
+                    if registry_key in self._registry:
+                        self._registry.pop(registry_key)
+                    else:
+                        raise KeyError(
+                            f"Registry key `{registry_key}` provided, but registry does not have this key.\n"
+                            f"Available keys in registry : {list(self._registry.keys())}"
+                        )
+
+        # Explicitly disable registry cache after reset
+        AccessMixin.set_access_enabled(access_enabled=False)
```

Comment:
should this be always or only for global reset?

Reply:
Always. Resets are first and last call, any call to if should indicate done with the registry.

Reply:
It doent delete the registry itself, just disables adding more things to it.